LibLancBlocking.cpp

```diff
@@ -0,0 +1,391 @@
+        lockout_evaluation_startTime = millis();
+        pin_status = inputState();
+        wait_high_startTime = micros();
+        while (pin_status)
+        {
+            if (lockout_evaluation_startTime - LANC_BLOCKED_CALL_TIMEOUT_MS < 0)
+            {
+                lockout_evaluation_startTime = millis();
+            } // Account for clock rollover
+            if (millis() - lockout_evaluation_startTime >= LANC_BLOCKED_CALL_TIMEOUT_MS)
+            {
+                LANC_FUNCTION_LOCKOUT = true;
+                ClearCommand();
+                return false;
+                // Time to disable the blocking until external issue is resolved.
+            }
+            pin_status = inputState();
+        }
+        // ***************
+
+        // Wait for input to move from LOW to HIGH
+        // ***************
+        lockout_evaluation_startTime = millis();
+        wait_low_startTime = micros();
+        while (!pin_status)
+        {
+            if (lockout_evaluation_startTime - LANC_BLOCKED_CALL_TIMEOUT_MS < 0)
+            {
+                lockout_evaluation_startTime = millis();
+            } // Account for internal clock rollover
+            if (millis() - lockout_evaluation_startTime >= LANC_BLOCKED_CALL_TIMEOUT_MS)
+            {
+                LANC_FUNCTION_LOCKOUT = true;
+                ClearCommand();
+                return false;
+                // Time to disable the blocking until external issue is resolved.
+            }
+            pin_status = inputState();
+        }
+        // ***************
+
+        // Evaluate if we have exceeded the total allowed time for this sync to run.  If exceeded lockout and return false.
+        long elapsed = millis() - syncTransmission_startTime;
+        if (elapsed >= (LANC_BLOCKED_CALL_TIMEOUT_MS))
+        {
+            LANC_FUNCTION_LOCKOUT = true;
+            ClearCommand();
+            return false;
+            // Time to disable the blocking until external issue is resolved.
+        }
+
+        // Evaluate if the transition state duration suggests we are at the end of the first start bit,
+        //  This indicates to us we are ready to begin transmitting data.  If so return the current time in microseconds.
+        pulse_transition_duration = wait_low_startTime - wait_high_startTime;
+        if (pulse_transition_duration > 3000)
+        {
+            bool_wait_for_start_bit = false;
+            return micros();
+        }
+    }
+    return micros();
+}
+
+void LancBlocking::delayUsWithStartTime(unsigned long startTime, unsigned long waitTime)
+{
+    while ((micros() - startTime) < waitTime)
+    {
+        // loop until the delay is waited
+    }
+}
+
+// This function simply waits for the next high->low->high transition,  its assumed you are calling this during the stop bits of the transaction.  No timing check outside of global lockout is evaulated.  Returns current time (us) once function passed
```

Comment:
Can you use doxygen style comments please? Those are standard and thus easy to understand by someone familiar with this. Also this allows to generate nice html documentation if needed